### PR TITLE
refactor: Variable indexing

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/debugging/macro_param_variable.cpp [hunk 1]
```diff
@@ -42,32 +42,22 @@ const std::string& macro_param_variable::get_value() const { return value_; }
...
 std::vector<variable_ptr> macro_param_variable::values() const
 {
     std::vector<std::unique_ptr<variable>> vals;
 
-    std::vector<size_t> child_index = index_;
+    std::vector<context::A_t> child_index = index_;
     child_index.push_back(0);
 
-    if (macro_param_.access_system_variable() && child_index.size() == 1)
+    if (auto index_range = macro_param_.index_range(index_); index_range)
```
[the remark this answers is not in the document]
Comment (slavek-kucera) [reply parser_library/src/debugging/macro_param_variable.cpp:54]: Well, it generates an empty vector, which is then shown in the DAP client as an array then expands into nothing. It looks fine to me.
From the technical point of view, everything seems to be well-defined, the only inefficiency is that we create a copy of child_index that is immediately freed in the case. I can move the checking up.

File: parser_library/test/common_testing.cpp [hunk 2]
```diff
@@ -158,9 +158,9 @@ std::optional<std::unordered_map<size_t, T>> get_var_vector_map(hlasm_context& c
 
     std::unordered_map<size_t, T> result;
     result.reserve(keys.size());
-    for (size_t i = 0; i < keys.size(); ++i)
+    for (auto key : keys)
```
[the remark this answers is not in the document]
Comment (slavek-kucera) [reply parser_library/test/common_testing.cpp:161]: Yes, but those are ints.

File: parser_library/src/context/hlasm_context.cpp [hunk 3]
```diff
@@ -651,23 +641,23 @@ std::vector<id_index> hlasm_context::whole_copy_stack() const
...
     if (scope->is_in_macro())
     {
         auto m_tmp = scope->this_macro->named_params.find(name);
         if (m_tmp != scope->this_macro->named_params.end())
-            return m_tmp->second;
+            return m_tmp->second.get();
     }
 
-    return var_sym_ptr();
+    return nullptr;
```
[the remark this answers is not in the document]
Comment (slavek-kucera) [reply parser_library/src/context/hlasm_context.cpp:660]: That is a check for nullptr... or am I missing something?

[the remark this answers is not in the document]
Comment (jirimosinger) [reply parser_library/src/context/hlasm_context.cpp:660]: No, you're not. Just no coffee, yet...

**Review by jirimosinger — APPROVED** [review]

**Pull request merged**